docs/concepts/endpoint.md

```diff
@@ -0,0 +1,117 @@
+---
+id: endpoint
+title: Endpoint
+---
+
+Endpoints in ZIO HTTP are defined using the `Endpoint` object's combinators, which provide a type-safe way to specify various aspects of the endpoint. For instance, consider defining endpoints for retrieving user information and user posts:
+
+```scala mdoc:silent
+import zio._
+import zio.http._
+import zio.http.endpoint.{Endpoint, EndpointExecutor, EndpointLocator, EndpointMiddleware}
+import zio.http.codec.{HttpCodec, PathCodec}
+import HttpCodec.query
+
+val auth = EndpointMiddleware.auth
+
+val getUser = Endpoint(Method.GET / "users" / int("userId")).out[Int]
+
+val getUserPosts = Endpoint(Method.GET / "users" / int("userId") / "posts" / int("postId"))
+  .query(query("name"))
+  .out[List[String]] @@ auth
+```
+
+In these examples, we use combinators like `Method.GET`, `int`, and `query` to define the HTTP method, URL path, path parameters, query parameters, and response types of the endpoints.
+
+### Middleware Application
```

Comment:
Why start with a header with level 3?
Its possible to start with level 2. something like this:

```md
## Level 2 (A Section)

## Level 2 (B Section)

### Level 3 (BA)

### Level 3 (BB)
```